### ISSUE TITLE
ocropus-gpageseg: Defective line splitting

### ISSUE THREAD

**Issue by wrznr** [tmbdev/ocropy#210]
## Expected Behavior
Simple running text should be consistently split into lines.

## Current Behavior
Currently working on data from the [Grenzboten project](https://www.suub.uni-bremen.de/ueber-uns/projekte/grenzboten/) together with @uvius. For some images, line splitting does not work. It is not clear why because very similar images are split correctly.

## Steps to Reproduce (for bugs)
1. Download test files: 

![179411_01 nrm](https://cloud.githubusercontent.com/assets/26707219/25614998/063f51b2-2f35-11e7-91ec-c11b55e4e5d5.png)
![179411_01 bin](https://cloud.githubusercontent.com/assets/26707219/25614999/063fa2c0-2f35-11e7-9c83-0cb354a42522.png)

2. Run `ocropus-gpageseg` on testfile(s).
3. Inspect results.

Test files have been created with `ocropus-nlbin`. Tested various command line parameter settings without success.

## Your Environment
* Python version: Python 2.7.3
* Git revision of ocropy:  commit 49c7f9ec943bda2c249b9027776d0e3de477b5ec
* Operating System and version: Linux lal 3.2.0-4-amd64 #1 SMP Debian 3.2.73-2+deb7u2 x86_64 GNU/Linux

**Comment by zuphilip** [comment]
Okay, I looked at the debug output with ` --debug` and it seems that the detected scale is too small (approximately half of the correct size):

![scale-default_lineseeds](https://cloud.githubusercontent.com/assets/5199995/25630565/e9cd66a2-2f6d-11e7-89c8-1b600c7348b9.png)

The disconnected (red) components are then creating the different lines.

If you increase that value by hand by setting the `--scale` parameter:
```
ocropus-gpageseg grenzboten.bin.png -n --debug --scale 30
```
then the output looks good:

![scale-30_lineseeds](https://cloud.githubusercontent.com/assets/5199995/25630660/2e3a6ac4-2f6e-11e7-95da-b37d53899262.png)

(Don't forget to remove all old images from the directory containing the lines.)

**Comment by wrznr** [comment]
Alright, thanks. That fixes the issue for the specific image (and many others). But if I set this parameter globally for the whole (pre-segmented) book, new problems arise with smaller (e.g. on-line images). Is there a known bug in the scale detection?

**Comment by zuphilip** [comment]
> new problems arise with smaller (e.g. on-line images)

I don't know what exactly you mean with "on-line images", but in general when you have to deal with font sizes which vary much (header vs. body text vs. footnote text), then ocropus has some problems and you might need some other steps.

> Is there a known bug in the scale detection?

Nothing I am aware of, but the example you provide looks like not an optimal guess from ocropus for the scale parameter. My guess is that for your test image the binarization will produce characters that are splitted into several connected components, and this influences the [estimation of the scale parameter](https://github.com/tmbdev/ocropy/blob/2e62b8e4fa2577f5f833bc54b63dc12617712b72/ocrolib/psegutils.py#L42-L50). I tried another binarization method here, and then the result seems also okay.

**Comment by wrznr** [comment]
Sorry @zuphilip. This is a typo and should be "one-line images" (i.e., images which cover only a single line). So it's not the varying font size but rather varying clipping sizes from the whole page image which cause the issues.

> I tried another binarization method here, and then the result seems also okay.

This is a great idea. I used `ocropus-nlbin` which seems the most obvious choice. From my experience, the tesseract line splitting is far superior to `ocropous-gpagseg` but this probably boils down to binarization.

Many thanks for your ongoing support!

**Comment by amitdo** [comment]
https://github.com/tmbdev/ocropy/blob/master/OLD/ocropus-sauvola

**Comment by zuphilip** [comment]
The scale estimation in ocropus for your example will produce this scalemap

![grenzboten-scalemap](https://cloud.githubusercontent.com/assets/5199995/26152568/b7ea9ff6-3b07-11e7-871a-20d6a232ded2.png)

As far as I understand the following happen then: For each of these boxes the algorithm continues to calculate the area and then take the square root (i.e. geometric mean of width and height). Overall the median of these numbers (without outliners) is then taken. Maybe in your example there are too many small connected components an/or the font is too narrow... 

( The corresponding Jupyter notebook is here: https://gist.github.com/zuphilip/e551ba6b733b5094749799651e4fbd3e )

Sauvola is one possibility and I the ocropus-nlbin has more parameters to try out. Moreover, it should be possible to mix some of the steps of Tesseract with some of the steps with Ocropus.

@wrznr Thank you for asking interesting questions!

**Comment by wrznr** [comment]
Indeed, using e.g. [scantailor](https://github.com/scantailor/scantailor) for binarization results in an almost error-free line splitting!  Only small one-line segments like page numbers and signature marks (which is probably to be expected) are not correctly processed. Significant step forward!

While this is great for me, is it a problem for ocropus (I.e. problems in the combination of `nlbin` and `gpageseg`)?
